components/search/types.ts

```diff
@@ -31,7 +34,7 @@ export type StateProps = {
     isPinnedPosts: boolean;
     isChannelFiles: boolean;
     filesSearchEnabled: boolean;
-    currentChannelId: string;
+    currentChannel: Channel;
```

Comment:
```suggestion
    currentChannel: Channel | undefined;
```

Reply:
This will be sufficient, so that typescript can add the `| undefined`

```suggestion
    currentChannel?: Channel;
```